src/users/containers/invite-user/invite-user.component.ts

```diff
@@ -19,50 +22,182 @@ import {AppConstants} from '../../../app/app.constants';
+  public jurisdictions: any[];
+  public juridictionSubscription: Subscription;
+  public resendInvite: boolean = false;
+  public pendingUserSubscription: Subscription;
+  public showWarningMessage: boolean;
+
+  public ngOnInit(): void {
+    this.showWarningMessage = false;
+    this.dispathAction(new fromStore.Reset(), this.store);
+    this.resendInvite = false;
     this.errors$ = this.store.pipe(select(fromStore.getInviteUserErrorMessage));
     this.errorsArray$ = this.store.pipe(select(fromStore.getGetInviteUserErrorsArray));
 
-    this.inviteUserForm = new FormGroup({
-      firstName: new FormControl('', Validators.required),
-      lastName: new FormControl('', Validators.required),
-      email: new FormControl('', [Validators.email, Validators.required]),
-      roles: new FormGroup({
-        'pui-case-manager': new FormControl(''),
-        'pui-user-manager': new FormControl(''),
-        'pui-organisation-manager': new FormControl(''),
-        'pui-finance-manager': new FormControl('')
-      }, checkboxesBeCheckedValidator())
+    this.inviteUserForm = this.initialiseUserForm();
+    this.backLink = this.getBackLink(null);
+    this.pendingUserSubscription = this.store.pipe(select(fromStore.getGetReinvitePendingUser)).subscribe(pendingUser => {
+      this.populateFormControl(pendingUser, this.inviteUserForm);
+      this.backLink = this.getBackLink(pendingUser);
     });
+
     this.juridictionSubscription = this.store.pipe(select(fromAppStore.getAllJurisdictions))
                                    .subscribe(value => this.jurisdictions = value,
                                    (error) => this.store.dispatch(new fromAppStore.LoadJurisdictionsFail(error)));
-    this.store.dispatch(new fromAppStore.LoadJurisdictions());
+    this.dispathAction(new fromAppStore.LoadJurisdictions(), this.store);
+
+    this.actions$.pipe(ofType(fromStore.INVITE_USER_FAIL_WITH_400)).subscribe(() => {
+      this.handleError(this.store, 400);
+    });
+    this.actions$.pipe(ofType(fromStore.INVITE_USER_FAIL_WITH_404)).subscribe(() => {
+      this.handleError(this.store, 404);
+    });
+    this.actions$.pipe(ofType(fromStore.INVITE_USER_FAIL_WITH_500)).subscribe(() => {
+      this.handleError(this.store, 500);
+    });
+    this.actions$.pipe(ofType(fromStore.INVITE_USER_FAIL_WITH_429)).subscribe(() => {
+      this.showWarningMessage = true;
+    });
+    this.actions$.pipe(ofType(fromStore.INVITE_USER_FAIL)).subscribe(() => {
+      this.store.dispatch(new fromAppStore.Go({ path: ['service-down'] }));
+    });
+  }
+
+  public handleError(store: Store<any>, errorNumber: number) {
+    const globalError = this.getGlobalError(errorNumber);
+    store.dispatch(new fromAppStore.AddGlobalError(globalError));
+    store.dispatch(new fromAppStore.Go({ path: ['service-down'] }));
+  }
+
+  public getGlobalError(error: number): GlobalError {
+    switch (error) {
+      case 400:
+        return this.get400Error();
```

Comment:
Not overly keen on this approach, as it means we'd have to have more functions per status code etc. How about a mapping instead?

Reply:
Done